script/hassfest/requirements.py

```diff
@@ -74,8 +81,45 @@ def validate(integrations: dict[str, Integration], config: Config):
         validate_requirements(integration)
 
 
+def validate_requirements_format(integration: Integration) -> bool:
+    """Validate requirements format.
+
+    Returns if an error was added.
+    """
+    start_errors = len(integration.errors)
+
+    for req in integration.requirements:
+        if " " in req:
+            integration.add_error(
+                "requirements",
+                f'Requirement "{req}" contains a space',
+            )
+            continue
+
+        pkg, sep, version = req.partition("==")
+
+        if not sep and integration.core:
+            integration.add_error(
+                "requirements",
+                f'Requirement {req} need to be pinned "<pkg name>==<version>".',
+            )
+            continue
+
+        if AwesomeVersion(version).strategy == AwesomeVersionStrategy.UNKNOWN:
+            integration.add_error(
+                "requirements",
+                f"Unable to parse package version ({version}) for {pkg}.",
+            )
+            continue
+
+    return len(integration.errors) > start_errors
+
+
 def validate_requirements(integration: Integration):
     """Validate requirements."""
+    if not validate_requirements_format(integration):
```

Comment:
Is this correct?
We should not proceed if the validation was successful?

Reply:
correct. Otherwise it would go ahead and install a bunch of stuff but if we already know the format is incorrect, let's get the user to fix that first.